hangman.js

```diff
@@ -1 +1,81 @@
-console.log("welcome to hangman");
\ No newline at end of file
+
+var hangman = {
+  word: "",
+  letter: "",
+  newWord: "",
+  welcome: alert("welcome to hangman"),
+  displayDashes: [],
+  counterAttempts: 6,
+
+  secretWord: function(){
+    this.word = prompt("Player 1 please type your secret word.", "Type word here.").toLowerCase();
+    return this.word;
+  },
+
+  //split word into array.
+  splitWord: function(){
+    this.newWord = this.word.split("");
+  },
+
+  //print the number of letters to screen OR display dashes.
+  blankSpaces: function(){
+    //for each letter in hangman.splitWord replace with a dash.
+    for(i=0; i<this.word.length; i++){
+      this.displayDashes.push("-");
+    }
+  },
+
+
+  //ask for guess
+  guessLetter: function(){
+    this.letter = prompt("Please guess a letter Player2").toLowerCase();
+  },
+
+
+  //iterate over word/array
+  compareToWord: function(){
+    while(this.counterAttempts > 0){
+      if(this.newWord.includes(this.letter)){
+        for(var i = 0; i<this.newWord.length; i++){
+          if(this.newWord[i] === this.letter){
+          this.displayDashes[i] = this.letter;
+          }
+        }
+        alert(this.displayDashes);
+          if(this.displayDashes.includes("-")){
+            this.guessLetter();
+            this.compareToWord();
+          }
+          else{
+            alert("You win!");
```

Comment:
You will need to close out of the `while` loop in this win block. Otherwise, it will keep alerting this infinitely. One idea is setting `counterAttemps` to 0, although I would try thinking of a semantically better solution.